### ISSUE TITLE
Compilation failed on RHEL 6

### ISSUE THREAD

**Issue by mgalmard** [modauthgssapi/mod_auth_gssapi#147]
Hi 

Don't know if it is the right place to post...

I am trying to compile the module for an apache 2.4 on RHEL 6 but I am running through a problem. 

Here is my configure syntax 
```
./configure --prefix=/usr/product/apache/httpd-2.4.25 --with-apxs=/usr/product/apache/httpd-2.4.25/bin/apxs --with-apr=/usr/product/apache/libs-2.4.25/bin/apr-1-config
```

It goes without any problem but when I try to make I get this : 
```
make[2]: entrant dans le répertoire « /dn-compil/ntiers/compil/sources/kerberos/mod_auth_gssapi-master/src »
/usr/product/apache/libs-2.4.25/build-1/libtool --silent  --tag=CC   --mode=compile gcc -std=gnu99 -DHAVE_CONFIG_H -I.  -I ./asn1c  -DLINUX -D_REENTRANT -D_GNU_SOURCE  -g -O2 -pthread    -I/usr/product/apache/httpd-2.4.25/include  -I/usr/product/apache/libs-2.4.25/include/apr-1  -fPIC -g -O2 -MT mod_auth_gssapi_la-mod_auth_gssapi.lo -MD -MP -MF .deps/mod_auth_gssapi_la-mod_auth_gssapi.Tpo -c -o mod_auth_gssapi_la-mod_auth_gssapi.lo `test -f 'mod_auth_gssapi.c' || echo './'`mod_auth_gssapi.c
mod_auth_gssapi.c: In function 'mag_create_dir_config':
mod_auth_gssapi.c:1293: error: 'struct mag_config' has no member named 'ccname_envvar'`
```

Any help would be much appreciated. 

Regards. 

N.B. I had to upgrade autoconf and bison to get that far.

**Comment by frozencemetery** [comment]
mgalmard <notifications@github.com> writes:

> * For information I had to upgrade autoconf and bison to get that far.

Well, we don't test on RHEL-6, as I'm sure you now know :)
Let me see if I can give more information:

> mod_auth_gssapi.c:1293: error: 'struct mag_config' has no member named 'ccname_envvar'`

This is gated on HAVE_CRED_STORE, which refers to the krb5 cred store
[extensions](http://k5wiki.kerberos.org/wiki/Projects/Credential_Store_extensions).
Which is to say: it needs krb5 > 1.11.

@simo5, we've guarded the instance in the definition with the #ifdef,
but none of the usages are guarded, and we use this variable in a few
places.  I think it's correct to move this variable out of the #ifdef?

As a related change, we may wish to consider making the cred store
extensions mandatory for newer releases.

**Comment by mgalmard** [comment]
Hi 

Thanks for the quick answer. I use krb5 1.10. As i really need to get this working on RHEL 6 I'll try to upgrade it as well and let you know.

Regards.

**Comment by frozencemetery** [comment]
@mgalmard I believe you should be able to move `ccname_envvar` in the struct definition in mod_auth_gssapi.h outside of the ifdef (i.e., down a line) and it will work, but I haven't tested.

**Comment by mgalmard** [comment]
@frozencemetery i upgraded the krb5 to 1.15 and manage to compile to the end and launch apache properly. 

Thanx again for your help :)

**Comment by simo5** [comment]
Fixed in #149
